File: basic/1007. 素数对猜想 (20).cpp
#include<iostream>
using namespace std;
int prime[100001];
void init(){
	fill(prime,prime+100001,1);
	prime[0]=prime[1]=0;
	for(int i=2;i<=1e5;++i){
		if(!prime[i])continue;
		for(int j=i*2;j<=1e5;j+=i)
			prime[j]=0;
	}
}
int main(){
	init();
	int n,cnt=0;
	cin>>n;
	for(int i=2;i<=n-2;++i){
		if(prime[i]&&prime[i+2])++cnt;
	}
	cout<<cnt;
	return 0;
}

File: basic/java/1007. 素数对猜想 (20).java
import java.util.LinkedList;
import java.util.Scanner;

public class Main {
	public static void main(String args[]) {
		Scanner in=new Scanner(System.in);
		int n=in.nextInt();
		in.close();
		LinkedList<Integer> prime=new LinkedList<Integer>();
		for(int i=2;i<=n;++i)
			if(isPrime(i))prime.add(i);
		int cnt=0;
		for(int i=0;i<prime.size()-1;++i)
			if(prime.get(i)-prime.get(i+1)==-2)++cnt;
		System.out.println(cnt);
	}
	public static boolean isPrime(int n) {
		for(int i=2;i<=Math.sqrt(n);++i)
			if(n%i==0)return false;
		return true;
	}
}
